presentation-api/controlling-ua/PresentationConnection_onclosed-manual.html

```diff
@@ -21,28 +21,36 @@
 <script>
   setup({explicit_timeout: true});
   var startPresentation = function () {
+
     document.getElementById('presentBtn').disabled = true;
-    async_test(function(t) {
+
+    promise_test(function (t) {
       var request = new PresentationRequest(presentationUrls);
-      request.start()
-        .then(function(connection) {
+
+      return request.start()
+        .then(function (connection) {
 
           // Enable timeout again, cause no user action is needed from here.
-          t.step_timeout(function() {
+          t.step_timeout(function () {
               t.force_timeout();
               t.done();
           }, 5000);
 
           assert_true(connection instanceof PresentationConnection, 'the connection is setup');
-          connection.onclose = t.step_func_done(function(evt) {
+
+          const watcher = new EventWatcher(t, connection, 'close');
+
+          watcher.wait_for('close').then(function () {
+            connection.close();
+            return watcher.wait_for('close');
```

Comment:
I did this because the API uses a similar example. See [here](https://github.com/w3c/testharness.js/blob/master/docs/api.md#promise-tests). There also the watcher is returned with all events to watch for.